test/common/stats/custom_stat_namespaces_impl_test.cc

```diff
@@ -0,0 +1,34 @@
+#include "source/common/stats/custom_stat_namespaces_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Stats {
+
+TEST(CustomStatNamespacesImpl, Registration) {
+  CustomStatNamespacesImpl namespaces;
+  const auto name = "foo";
```

Comment:
const std::string here. I am actually not sure what type the C++ compiler would invent for a C string literal assigned to an auto. Maybe a `char*` or `char[]`?

Reply:
thank you for bearing with me - changed to std::string.